File: src/services/compliance.ts
export interface ComplianceStandard {
  name: string;
  score: number | null;
  status: "Compliant" | "Needs Review" | "Needs Improvement" | "Not Evaluated";
  reason: string;
}

export interface ComplianceResponse {
  overall: number | null;
  standards: ComplianceStandard[];
  message?: string | null;
}

const API_URL = import.meta.env.VITE_API_URL ?? "http://127.0.0.1:8000";

export async function getCompliance(): Promise<ComplianceResponse> {
  const response = await fetch(`${API_URL}/compliance`);
  if (!response.ok) throw new Error("Unable to load compliance assessment.");
  return response.json();
}

File: backend/app/schemas/compliance.py
"""Pydantic models returned by the Compliance Center endpoint."""

from typing import Literal
from typing import Optional

from pydantic import BaseModel, Field


class ComplianceStandard(BaseModel):
    """Compliance assessment for a single industrial standard."""

    name: str = Field(..., min_length=1)
    score: Optional[int] = Field(None, ge=0, le=100)
    status: Literal["Compliant", "Needs Review", "Needs Improvement", "Not Evaluated"]
    reason: str = Field(..., min_length=1)


class ComplianceResponse(BaseModel):
    """Aggregated Compliance Center assessment."""

    overall: Optional[int] = Field(None, ge=0, le=100)
    standards: list[ComplianceStandard] = Field(default_factory=list)
    message: str | None = None
